packages/rocketchat-api/server/v1/channels.js

```diff
@@ -967,3 +967,54 @@ RocketChat.API.v1.addRoute('channels.moderators', { authRequired: true }, {
 		});
 	},
 });
+
+RocketChat.API.v1.addRoute('channels.addLeader', { authRequired: true }, {
+        post() {
+                const findResult = findChannelByIdOrName({ params: this.requestParams() });
+
+                const user = this.getUserFromParams();
+
+                Meteor.runAsUser(this.userId, () => {
+                        Meteor.call('addRoomLeader', findResult._id, user._id);
+                });
+
+                return RocketChat.API.v1.success();
+        },
+});
+
+RocketChat.API.v1.addRoute('channels.removeLeader', { authRequired: true }, {
+        post() {
+                const findResult = findChannelByIdOrName({ params: this.requestParams() });
+
+                const user = this.getUserFromParams();
+
+                Meteor.runAsUser(this.userId, () => {
+                        Meteor.call('removeRoomLeader', findResult._id, user._id);
+                });
+
+                return RocketChat.API.v1.success();
+        },
+});
+
+RocketChat.API.v1.addRoute('channels.setEncrypted', { authRequired: true }, {
+        post() {
+                if (typeof this.bodyParams.encrypted === 'undefined') {
+                        return RocketChat.API.v1.failure('The bodyParam "encrypted" is required');
+                }
+
+                const findResult = findChannelByIdOrName({ params: this.requestParams() });
+
+                if (findResult.encrypted === this.bodyParams.encrypted) {
+                        return RocketChat.API.v1.failure('The channel encrypted is the same as what it would be changed to.');
+                }
+
+                Meteor.runAsUser(this.userId, () => {
+                        Meteor.call('saveRoomSettings', findResult._id, 'encrypted', this.bodyParams.encrypted);
```

Comment:
Before saving should we verify this is a bool? This way we don't risk an object being passed

Reply:
Hi,
(Sorry I was in vacations)
How we can check it ?
Regards